sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_client_manager.py

```diff
@@ -323,28 +334,80 @@ def __init__(
                 )
             )
             self._setup_failover_endpoints()
+            if load_balance:
+                random.shuffle(self._replica_clients)
             return
         raise ValueError("Please pass either endpoint and credential, or a connection string with a value.")
 
+    def get_next_client(self) -> Optional[_ConfigurationClientWrapper]:
+        """
+        Get the next client to be used for the request. find_active_clients needs be called before this method or None
+        client will be returned. If load_balance isn't enabled the first active client will be returned, which is the
+        provided endpoint unless it has failed. If load_balance is enabled the next client in the list will be returned.
+
+        :return: The next client to be used for the request.
+        """
+        if not self._active_clients:
+            return None
+        if not self._load_balance:
+            return self._active_clients[0]
+        next_client = self._active_clients.pop(0)
+        self._last_active_client_name = next_client.endpoint
+        return next_client
+
+    def has_next_client(self) -> bool:
+        """
+        Check if there is another client to be used for the request. find_active_clients needs be called before this
+        method or False will be returned.
+
+        :return: True if there is another client to be used for the request, False otherwise.
+        :rtype: bool
+        """
+        return bool(self._active_clients)
+
+    def find_active_clients(self):
+        """
+        Figures out the current active clients, if load_balance is enabled the start of the list will be the one after
+        the last active client used.
+        """
+        active_clients = [client for client in self._replica_clients if client.is_active()]
+
+        self._active_clients = active_clients
+        if not self._load_balance or len(self._last_active_client_name) == 0:
+            return
+        for i, client in enumerate(active_clients):
+            if client.endpoint == self._last_active_client_name:
+                swap_point = (i + 1) % len(active_clients)
```

Comment:
Just curious, are we trying to re-arrange this to continue from where the last active client was before refresh?

Reply:
Yes, that's the logic we use for load balancing. @amerjusupovic